### DISCUSSION TITLE
how to change code block height and width in quarto presentation

### DISCUSSION THREAD

**Discussion by lizhiwei1994** [quarto-dev/quarto-cli#2073] [Q&A]
I am using quarto to make slides based on revealjs.
I can not find options that can change code block height and width in the [Guide](https://quarto.org/docs/presentations/revealjs/).

In the example beleow maybe we can set like this although it is a wrong way.
```
    ```{r code_block code-block-width="135px" code-block-height="200px"}
    some code here
    ```
```
    

An example is below.

```
    ---
    title: "Untitled"
    format: revealjs
    editor: visual
    ---
    
    ## 
    text text text text text
    
    ::: columns 
    
    ::: {.column width="50%"}
    ```{r code_block}
    #| eval: false
    #| echo: true
    
    some code here
    ```
    :::
    
    ::: {.column width="50%"}
    
    ```{r}
    a = 1
    b = 2
    plot(a, b)
    ```
    
    ```{r}
    a = 1
    b = 2
    plot(a, b)
    ```
    
    :::
    :::    
```

**Comment by melissavanbussel** [comment]
One way to achieve this would be to include a custom css file, and then edit the height and width there. 

Contents of .qmd file:

~~~
---
title: "Untitled"
format: revealjs
editor: visual
css: style.css
---

## Code

When you click the **Render** button a presentation will be generated that includes both content and the output of embedded code. You can embed code like this:

```{r}
#| echo: true
#| eval: false
1 + 1
```  
~~~

In the `style.css` file: 

```
.reveal pre.sourceCode code {
	height: 200px;
}
```

Rendering the file results in the following: 

<img width="960" alt="image" src="https://user-images.githubusercontent.com/28693927/194973863-6ce40f49-f547-4e10-aff4-74f2892220ac.png">

**Reply by RaymondBalise** [reply]
This is super helpful.  Is there a way to apply this solution to only one code chunk?

**Reply by mcanouil** [reply]
Add an ID, and use that to better target your CSS rule.

**Reply by RaymondBalise** [reply]
Thank you for the fast response.  I apologize for being clueless but unfortunately I don't know how to add an ID to a code chunk. I was able to add a div and target it like this:

````quarto
```{css}
.the_plot_code .cell .sourceCode.cell-code pre code.sourceCode.r {
  height: 200px;
}
```

::: {.the_plot_code}
```{r}
#| label: the-plot-code
#| echo: true
#| eval: true

```
````

Is this what you had in mind?

**Reply by mcanouil** [reply]
`label` is the ID of a code cell.
Note that you should prefer dash instead of underscores for IDs/labels.

**Reply by RaymondBalise** [reply]
Can you please show me an example  I tried this:

````
```{css}
.the-plot-code-x .cell .sourceCode.cell-code pre code.sourceCode.r {
  height: 200px;
  color: pink;
}
```

::: {.the_plot_code}
```{r}
#| label: the-plot-code-x
#| echo: true
#| eval: true

# stuff
```
:::
````

**Reply by mcanouil** [reply]
Here:

<table>
<tr><th>Input</th><th>Output</th></tr>
<tr><td>

````qmd
---
format: revealjs
execute:
    eval: false
    echo: true
include-in-header:
    - text: |
        <style>
            #my-code pre.sourceCode {
                height: 150px;
                background: #b22222;
            }
        </style>
---

## Slide 1

```{python}
#| label: my-code
print("Hello, World!")
```

Some text

```{python}
#| label: not-my-code
print("Hello, World!")
```

````

</td><td>

<img width="730" alt="image" src="https://github.com/user-attachments/assets/971308e6-17c3-46ab-a252-be90ac071e0a" />

</td></tr>
</table>

**Reply by DaniMori** [reply]
This seems to work for python but not R chunks, and only for the height, not the background 😞

**Reply by mcanouil** [reply]
Any reproducible example showing that it's not working?
Note that even if the above does not work as is, the same principle apply, so you could still use your browser to make the right CSS rule.

**Reply by cscheid** [reply]
This is a Quarto bug - the id used in code cell outputs is not consistent from Jupyter to knitr.

**Reply by mcanouil** [reply]
Here, following `knitr` documentation, here is one way amongst many:

<table>
<tr><th>Input</th><th>Output</th></tr>
<tr><td>

````qmd
---
format: revealjs
execute:
    eval: true
    echo: true
include-in-header:
    - text: |
        <style>
            #my-code pre.sourceCode {
                height: 150px;
                background: #b22222;
            }
        </style>
---

## Slide 1

```{r}
#| attr-source: "id=my-code"
print("Hello, World!")
```

Some text

```{r}
#| attr-source: "id=not-my-code"
print("Hello, World!")
```
````

</td><td>

<img width="715" alt="image" src="https://github.com/user-attachments/assets/57f898d1-1a6b-4f65-b19f-c588579b086e" />

</td></tr>
</table>

**Reply by DaniMori** [reply]
> Any reproducible example showing that it's not working? Note that even if the above does not work as is, the same principle apply, so you could still use your browser to make the right CSS rule.

Sorry, didn't provide a Reprex bc I just attempted the exact same doc but changing the chunks to R instead of Python.

But the latter solution worked, yay!! (Only, just for the height for me, as in the case with Python)

**Comment by camilogarciabotero** [comment]
There is an alternative if you want to make the code chunk and output side by side:

```{R}
#| output-location: column-fragment
#| echo: true

penguins |>
  filter(species == "Adelie") |>
  select(species, island, sex)
```
<img width="1023" alt="Screen Shot 2022-10-11 at 11 36 27 AM" src="https://user-images.githubusercontent.com/25586362/195149588-b806b06f-ebfa-4440-be2e-aceadf8af86b.png">

**Comment by mcanouil** [comment]
FYI
- #4943
